### PR TITLE
adds filter for url pattern

### PR DESCRIPTION
**Pull request by wheresrhys**
@GlynnPhillips @tom-parker

### CODE REVIEW

**Comment by tom-parker** [comment]
This looks good, can we keep it the same format as the other filters though?

E.g
```
abTestSetup: {
    range: 10,
    offset: 80,
    filters: [
        {
            type: "matchesPath",
            value: "/dummy-path"
        }
    ]
}
```

Also is it matching against `req.get('filter-by-url')` rather than the actual request path? I might have missed something here

**Comment by wheresrhys** [comment]
The flags-api (where I started all this work) has no tests for the filters property, so I didn't know it existed. Not documented anywhere either :rage:. Though I agree it makes sense to match that format so will refactor

**Comment by tom-parker** [comment]
There are some brief docs here - https://ammit.ft.com/dev-docs/

**Comment by wheresrhys** [comment]
for devs arguably they should be in the readme aswell/instead, or at least linked to from the README

**Comment by GlynnPhillips** [comment]
I think theres a lot of confusion caused by how tightly coupled flags and ammit are. Part of me thinks we should remove reference in the flags README to anything about properties to do with AB tests and just link to the ammit repo for that info which could also link to the docs site.

But I am not sure if this would cause more confusion when developing a feature and reading about it in the ammit README but having to implement it in the flags repo.

I would really like to spend some time sorting a lot of this because at the moment the docs (README and ammit.ft.com) don't really serve their purpose to devs or non devs. Finding the time though is difficult.

**Comment by wheresrhys** [comment]
After September it would be good to finally have that defining, building and rolling out of flags v2, and a tidy up of all the stuff that hangs off them

**Comment by wheresrhys** [comment]
How about if ammit provides a /validate end point, which encapsulates the AB validation rules and is called by the flags api? @GlynnPhillips

**Comment by GlynnPhillips** [comment]
@wheresrhys Yeah this is something we have thought about. If we get some time it would be a great start to the de-coupling.

File: server/filters/matches-path.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const pathToRegexp = require('path-to-regexp');
+
+const compiledPathCache = {};
+const compiledRegexCache = {};
+
+module.exports = (test, filter, user, req) => {
+
+	const url = req.get('filter-by-url');
```
Comment (tom-parker) [inline server/filters/matches-path.js]: Where is this header getting set? We need to make sure it's present when Spoor calls Ammit too.

**Comment by wheresrhys** [comment]
closing i favour of doing something similar in preflight that won't dilute the cache

**Pull request closed without merging**